iree/turbine/kernel/boo/fusion/subgraph.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2025 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from copy import deepcopy
+from typing import Sequence, Set, List, Dict, Tuple
+
+import torch
+from torch.fx.subgraph_rewriter import replace_pattern
+from torch.fx.graph import Graph
+from torch.fx.graph_module import GraphModule
+from torch.fx.node import Node
+
+from .schema import FusionSchema, OpFusionSpec
+
+
+def extract_fusion_subgraph_modules(
+    src_gm: GraphModule, fusion_schema: FusionSchema
+) -> Tuple[Sequence[GraphModule], Sequence[Dict[Node, Node]]]:
+    """Traverses src_gm nodes in order. When a node matches a root op in the fusion_schema,
+    A new subgraph is created with the root op in addition to any adjacent nodes matching the schema.
+
+    All subgraphs found this way are returned in a list.
+    The corresponding mappings from src_gm nodes -> subgraph nodes is also returned as a list.
+    """
```

Comment:
What prevents us from using the [SubgraphMatcher](https://github.com/pytorch/pytorch/blob/8597d37536ef11bdf6b0a539ab79af876e1c92f6/torch/fx/passes/utils/matcher_utils.py#L51)?